### DISCUSSION TITLE
Material Model of RBC

### DISCUSSION THREAD

**Discussion by blchrisum** [UvaCsl/HemoCell#82] [General discussion]
Hello all, 

I want to make changes to the RBC material model of HemoCell. 
I followed the discussion #21 and read the paper [https://www.frontiersin.org/articles/10.3389/fphys.2017.00563/full](url) about the various parameters in the "rbcHighOrderModel" model. 

But I'm still not sure on how to implement a new material model. 

1. **As a first step** I want to simulate **rigid RBCs** in order to compare this simulation with a spherical particle simulation of another simulation code
- Is there a way to do this efficently ? e.g. by not solving the material model equations and therefore the deformation of RBC?
- Or should i find very large values for the constants of the material model that are somehow numerically stable ?
- I already tried to skip the material update by increasing the "stepMaterialEvery" variable in the config file to a value bigger then the overall iterations, but this was not successful

 
2.  **As a second step** i want to find a set of constants that describe RBCs that can deform but at the same time can withstand very high shear rates (supraphysiological). 

- Do you think it is possible to find such constants with the current implementation ?
- Are these constants found by trial and error in the [examples/oneCellShear](https://github.com/UvaCsl/HemoCell/tree/master/examples/oneCellShear) and [examples/stretchCell](https://github.com/UvaCsl/HemoCell/tree/master/examples/stretchCell) examples or is there a better approach ?
- I was also thinking of introducing some kind of deformation threshold after which the mechanical equations are not solved any more. This would result in deformed RBC that are very stiff after reaching a certain deformation value. What do you think about that ?

**Comment by gzavo** [comment]
Hi @blchrisum,

1. The cells are coupled using the immersed boundary method which is inefficient at representing rigid objects.
You can increase the material coefficients to make an object more stiff, but then you will probably also need to decrease the time-step proportionately to keep the simulation stable. 
An additional helper solution can be to use the internal links (in the cell xml file) to add more "structural elements" to increase rigidity.
The `stepMaterialEvery` flag is just a numerical tool to trade accuracy for performance, it will not change the mechanical model. If you put it too high the coupling between the fluid and the cells will not work anymore.

2. As you see from 1) making the cell stiffer is not necessarily trivial. It is possible that a different set of parameters exists that is  more stable. The current set is found by narrowing the options down by sensitivity analysis ( https://doi.org/10.1615/Int.J.UncertaintyQuantification.2020033186 ), and then doing a search in the remaining parametric space.
As a possibility, there are other fluid-structure couplings that are more stable in this regard (e.g., Noble-Torczynski https://doi.org/10.1142/S0129183198001084 ), that in theory can be implemented in HemoCell. These, however, often have other drawbacks.

I think it is an interesting idea to create a material model that in low shear has the current model, and in higher shear it switches to a stiffer linear model that can stay more stable.
